File: 02_08_2024_Minimum_Swaps_to_Group_All_1's_Together_II.py
class Solution:
    def minSwaps(self, nums: List[int]) -> int:
        k = sum(nums)
        n = len(nums)
        cnt = sum(nums[:k])
        mx = cnt
        for i in range(k, n + k):
            cnt += nums[i % n] - nums[(i - k + n) % n]
            mx = max(mx, cnt)
        return k - mx

File: 02_08_2024_Minimum_Swaps_to_Group_All_1's_Together_II.java
class Solution {
    public int minSwaps(int[] nums) {
        int k = Arrays.stream(nums).sum();
        int n = nums.length;
        int cnt = Arrays.stream(Arrays.copyOfRange(nums, 0, k)).sum();
        int mx = cnt;
        for (int i = k; i < n + k; ++i) {
            cnt += nums[i % n] - nums[(i - k + n) % n];
            mx = Math.max(mx, cnt);
        }
        return k - mx;
    }
}

